test/jdk/javax/swing/JButton/HtmlButtonImageTest/HtmlButtonImageTest.java

```diff
@@ -91,14 +83,20 @@ public static void main(String[] args) throws Exception {
             Color botClr = robot.getPixelColor(point.x, point.y + (SQUARE_HEIGHT/2) - PIXEL_BUFFER);
 
             testImageCentering(leftClr, rightClr, topClr, botClr);
+        } catch (ClassNotFoundException | InstantiationException | IllegalAccessException e) {
+            throw new RuntimeException("Unsupported LookAndFeel: " + e);
+        } catch (IOException e) {
+            //save image for troubleshooting
+            BufferedImage failImg = robot.createScreenCapture(new Rectangle(point.x - BUTTON_WIDTH / 2, point.y - BUTTON_HEIGHT / 2, BUTTON_WIDTH, BUTTON_HEIGHT));
+            ImageIO.write(failImg, "png", new File(testDir + "/fail_square.png"));
+            throw new RuntimeException("Failed image generation: " + e);
```

Comment:
This doesn't make sense. I mean `IOException` can be thrown from try-block only from `generateImage` method. In this case, there's nothing to display on screen and therefore nothing to capture. You should not catch `IOException` at all.

You should create the screenshot if the conditions you verify aren't met. The screenshot should be taken inside `if (!checkRedness(c))` block before you throw the `RuntimeException` to indicate failure.

Reply:
I believe I fixed the additional changes requested in the next commit. I understand what you meant with main handling exceptions now.